flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/FlinkFilterJoinRule.java

```diff
@@ -386,6 +387,21 @@ private void pushFiltersToAnotherSide(
         }
     }
 
+    private boolean isSuitableFilterToPush(RexNode filter, JoinRelType joinType) {
+        if (filter.isAlwaysTrue()) {
+            return false;
+        }
+        // For left/right outer join, we cannot push down IS_NULL filter to other side. Take left
+        // outer join as an example, If the join right side contains an IS_NULL filter, while we try
+        // to push it to the join left side and the left side have any other filter on this column,
+        // which will conflict and generate wrong plan.
+        if ((joinType == JoinRelType.LEFT || joinType == JoinRelType.RIGHT)
```

Comment:
if we need to check full outer join here?

Reply:
Hi, @herunkang2018. For method FlinkFilterJoinRule#pushFiltersToAnotherSide, we only support Inner/left/Right join. So there is no need to judge the full outer join here.

Reply:
for left outer join, only the equal conditions from right side can be push to the left side. and vice versa for right outer join.

please add some non equal conditions (such as a2 <> 10 ) in the IT case

Reply:
> for left outer join, only the equal conditions from right side can be push to the left side. and vice versa for right outer join.
> 
> please add some non equal conditions (such as a2 <> 10 ) in the IT case

Done!

Reply:
additional, less(or equal) than, greater (or equal) than  a literal can also be supported

Reply:
please also take care about `a2 = null`